FLE Basis GPU tests

The current tests for the FLE Basis (#693) test combinations of size = `32, 33` and epsilons = `1e-4, 1e-7, 1e-10, 1e-14`

For the case `1e-7` case, the `cufinufft` tests are failing to be within the desired precision. Specifically, the following tests fail 

- `testFastVDense_T`
- `testFastVDense`
- `testEvaluateExpand`

Moving onto Decaf to debug, we have the following output:

```
FAILED tests/test_FLEbasis2D.py::TestFLEBasis2D::testFastVDense_T[33-1e-07] - assert 1.0770913896475136e-07 < 1e-07
FAILED tests/test_FLEbasis2D.py::TestFLEBasis2D::testFastVDense[32-1e-07] - assert 8.535542528013261e-06 < 1e-07
FAILED tests/test_FLEbasis2D.py::TestFLEBasis2D::testFastVDense[33-1e-07] - assert 3.0947756503646876e-05 < 1e-07
FAILED tests/test_FLEbasis2D.py::TestFLEBasis2D::testEvaluateExpand[32-1e-07] - RuntimeError: Unable to converge!
FAILED tests/test_FLEbasis2D.py::TestFLEBasis2D::testEvaluateExpand[33-1e-07] - RuntimeError: Unable to converge!
```

Another run:

```
FAILED tests/test_FLEbasis2D.py::TestFLEBasis2D::testFastVDense_T[32-1e-07] - assert 1.2307842145582195e-06 < 1e-07
FAILED tests/test_FLEbasis2D.py::TestFLEBasis2D::testFastVDense_T[33-1e-07] - assert 3.27830082146523e-06 < 1e-07
FAILED tests/test_FLEbasis2D.py::TestFLEBasis2D::testFastVDense[32-1e-07] - assert 3.683104864109316e-07 < 1e-07
FAILED tests/test_FLEbasis2D.py::TestFLEBasis2D::testFastVDense[33-1e-07] - assert 7.962711500110564e-06 < 1e-07
FAILED tests/test_FLEbasis2D.py::TestFLEBasis2D::testEvaluateExpand[32-1e-07] - RuntimeError: Unable to converge!
FAILED tests/test_FLEbasis2D.py::TestFLEBasis2D::testEvaluateExpand[33-1e-07] - RuntimeError: Unable to converge!
```

The "Unable to converge" comes from the `expand()` method.

Clearly the tests are pretty close but drop out of the desired range non deterministically

--- a/tests/test_FLEbasis2D.py
+++ b/tests/test_FLEbasis2D.py
@@ -21,10 +21,6 @@ def show_fle_params(basis):
     return f"{basis.nres}-{basis.epsilon}"
 
 
-def gpu_ci_skip():
-    pytest.skip("1e-7 precision for FLEBasis2D")
-
-
 fle_params = [
     (32, 1e-4),
     (32, 1e-7),
@@ -80,8 +76,6 @@ class TestFLEBasis2D(UniversalBasisMixin):
 
     # check closeness guarantees for fast vs dense matrix method
     def testFastVDense_T(self, basis):
-        if backend_available("cufinufft") and basis.epsilon == 1e-7:
-            gpu_ci_skip()
 
         dense_b = basis._create_dense_matrix()
 
@@ -97,8 +91,6 @@ class TestFLEBasis2D(UniversalBasisMixin):
         assert relerr(result_dense.T, result_fast) < (self.test_eps * basis.epsilon)
 
     def testFastVDense(self, basis):
-        if backend_available("cufinufft") and basis.epsilon == 1e-7:
-            gpu_ci_skip()
 
         dense_b = basis._create_dense_matrix()
 
@@ -120,8 +112,6 @@ class TestFLEBasis2D(UniversalBasisMixin):
         raises=RuntimeError,
     )
     def testEvaluateExpand(self, basis):
-        if backend_available("cufinufft") and basis.epsilon == 1e-7:
-            gpu_ci_skip()
 
         # compare result of evaluate() vs more accurate expand()
         # get sample coefficients
@@ -135,8 +125,6 @@ class TestFLEBasis2D(UniversalBasisMixin):
 
 @pytest.mark.parametrize("basis", test_bases_match_fb, ids=show_fle_params)
 def testMatchFBEvaluate(basis):
-    if backend_available("cufinufft") and basis.epsilon == 1e-7:
-        gpu_ci_skip()
 
     # ensure that the basis functions are identical when in match_fb mode
     fb_basis = FBBasis2D(basis.nres, dtype=np.float64)
@@ -170,8 +158,6 @@ def testMatchFBDenseEvaluate(basis):
 
 @pytest.mark.parametrize("basis", test_bases_match_fb, ids=show_fle_params)
 def testMatchFBEvaluate_t(basis):
-    if backend_available("cufinufft") and basis.epsilon == 1e-7:
-        gpu_ci_skip()
 
     # ensure that coefficients are the same when evaluating images
     fb_basis = FBBasis2D(basis.nres, dtype=np.float64)

